### ISSUE TITLE
CSpinBox is severly broken

### ISSUE THREAD

**Issue by JayFoxRox** [citra-emu/citra#1725]
The "spinbox" used in the framebuffer debugger is very broken.

* When entering hex letters in lower case ("a-f") they are visible but do not update the returned value.
* Furthermore it is (sometimes?) possible to enter more digits than available and the spinbox will return the rightmost values only. I'm not sure when this happens

These are actually 2 seperate issues, however, I think a meta-issue is fine here because we should consider using another class or we'll have some serious bugfixing to do..
At this point nobody should trust the values entered into the spinbox because the accepted value may be very different. I noticed this when I wanted to look at textures but I was able to show 2 different textures using the same address.

To give you a bit of context: the class is only used for address entry using a hex-base in the citra codebase (afaik), yet it implements signed numbers, different bases etc.
It should be fairly straightforward to replace it with a hex-spinbox (example: http://doc.qt.io/qt-5/qtwidgets-widgets-spinboxes-example.html has one, I didn't try it yet though)

**Comment by wwylele** [comment]
I just played with the spin box, and suddenly got a segmentation fault... this can be included in this issue.

**Comment by JayFoxRox** [comment]
No, that's a seperate issue in the framebuffer dialog which is fixed in one of my branches (which replaces the widget altogether).

**Comment by inputfail** [comment]
Does the strikethrough indicate that you already tried the hex QSpinBox from that document?

Also, forgive me, but you mentioned something about a return type in the IRC that I forgot?

**Comment by JayFoxRox** [comment]
QSpinBox uses a signed int (32 bit) datatype (-0x80000000 to 0x7FFFFFFF), however, we need 32 bit unsigned (for memory addresses; 0x00000000 to 0xFFFFFFFF).

Just create a new widget for testing or add it on the main window?
The debug widgets will only be enabled on debug breakpoints (open the "Pica breakpoints" debug widget and set a breakpoint, then wait until execution stops). However, as the framebuffer viewer is very crashy I recommend just creating your own widget for testing.
